mono/utils/mono-error.c

```diff
@@ -819,6 +828,51 @@ Convert this MonoError to an exception if it's faulty or return NULL.
 The error object is cleant after.
 */
 
+MonoExceptionHandle
+mono_error_convert_to_exception_handle (MonoError *target_error)
+{
+	ERROR_DECL (error);
+
+	HANDLE_FUNCTION_ENTER ()
+
+	MonoExceptionHandle ex = MONO_HANDLE_NEW (MonoException, NULL);
+
+	/* Mempool stored error shouldn't be cleaned up */
+	g_assert (!is_boxed ((MonoErrorInternal*)target_error));
+
+	if (mono_error_ok (target_error))
+		goto exit;
+
+	MONO_HANDLE_ASSIGN(ex, mono_error_prepare_exception_handle (target_error, error));
+	if (!mono_error_ok (error)) {
+		ERROR_DECL_VALUE (second_chance);
+		/*Try to produce the exception for the second error. FIXME maybe we should log about the original one*/
+		MONO_HANDLE_ASSIGN(ex, mono_error_prepare_exception_handle (error, &second_chance));
+
+		g_assert (mono_error_ok (&second_chance)); /*We can't reasonable handle double faults, maybe later.*/
+		mono_error_cleanup (error);
+	}
+	mono_error_cleanup (target_error);
+
+exit:
+	HANDLE_FUNCTION_RETURN_REF (MonoException, ex)
+}
+
+#if 0 // FIXMEcoop remove or refactor
```

Comment:
Why this dead code? Can't we just replace existing `mono_error_convert_to_exception`, or the other way of implementing `mono_error_convert_to_exception_handle` in terms of `mono_error_convert_to_exception`

Reply:
I think the preferred way is "native" is "handle" and "raw wraps it" but I was going for the safer change for now, just copy/paste/edit. The if 0 is what I think is ideal/smaller, but not worth short term risk, can fit it later.